slackbot.go

```diff
@@ -0,0 +1,129 @@
+*/
+
+package main
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/gravitational/logrus"
+	"github.com/gravitational/trace"
+
+	pagerduty "github.com/PagerDuty/go-pagerduty"
+	"github.com/shomali11/slacker"
+)
+
+// Init is called upon Bot creation (first startup)
+func Init(config *Config) {
+	log.Println("Connected!")
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var opts pagerduty.GetScheduleOptions
+	if schedule, err := client.GetSchedule(config.PagerDuty.Schedule, opts); err != nil {
+		trace.Wrap(err)
+	} else {
+		log.Println("Configured schedule is \"" + schedule.Name + "\" with ID: " + config.PagerDuty.Schedule)
+	}
+}
+
+// Err function is used to handle all Error reported by the Bot
+func Err(err string) {
+	fmt.Println(err)
+}
+
+// Emergency is used to open Emergency Incidents on PagerDuty
+func Emergency(request slacker.Request, response slacker.ResponseWriter, config *Config) {
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var scheduleOpts pagerduty.GetScheduleOptions
+	if schedule, err := client.GetSchedule(config.PagerDuty.Schedule, scheduleOpts); err != nil {
+		trace.Wrap(err)
+	} else {
+		log.Println("Opening incident on schedule \"" + schedule.Name +
+			"\"/" + config.PagerDuty.Schedule)
+	}
+
+	newIncident := pagerduty.CreateIncidentOptions{
+		Type: "incident",
+		Title: "Incident opened by " + config.CustomerName +
+			" via Slack/@" + config.Slack.BotUsername,
+	}
+
+	newIncidentBody := pagerduty.APIDetails{
+		Type:    "incident_body",
+		Details: request.Param("msg"),
+	}
+	newIncident.Body = newIncidentBody
+
+	newIncidentService := pagerduty.APIReference{
+		Type: "service_reference",
+		ID:   config.PagerDuty.Service,
+	}
+	newIncident.Service = newIncidentService
+
+	createIncidentOpts := pagerduty.CreateIncident{
+		Incident: newIncident,
+	}
+
+	if incident, err := client.CreateIncident(config.PagerDuty.FromEmail, &createIncidentOpts); err != nil {
+		errText := "There was an error while creating a new incident created, please try again and report the following error" + err.Error()
+		fmt.Println(errText)
+		response.Reply(errText)
+	} else {
+		incidentURL := config.PagerDuty.Link + "/incidents/" + incident.Id
+		log.Println("Incident created by " + config.CustomerName +
+			" via @" + config.Slack.BotUsername +
+			" > " + incidentURL)
+		response.Reply("Incident created successfully, " +
+			"please refer to incident " + incidentURL)
+
+	}
+}
+
+// Default function handles all messages that won't match the other Commands
+func Default(request slacker.Request, response slacker.ResponseWriter, config *Config) {
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var opts pagerduty.ListOnCallUsersOptions
+	opts.Since = time.Now().UTC().Format(time.RFC3339)
+	opts.Until = time.Now().UTC().Add(time.Minute * 1).Format(time.RFC3339)
+	if onCallUserList, err := client.ListOnCallUsers(config.PagerDuty.Schedule, opts); err != nil {
+		trace.Wrap(err)
+	} else {
```

Comment:
1. Delete else here. Use `response.Reply(textError)` to reply with error in the line above.
2. `onCallUserList` will be unavailable outside of if scope

Reply:
added response though :+1: